indexer-balances/src/db_adapters/balance_changes.rs

```diff
@@ -581,121 +560,35 @@ fn get_deltas(
     ))
 }
 
-async fn get_balance_retriable(
-    account_id: &near_indexer_primitives::types::AccountId,
-    block_hash: &near_indexer_primitives::CryptoHash,
-    balance_cache: &crate::BalanceCache,
-    json_rpc_client: &near_jsonrpc_client::JsonRpcClient,
-) -> anyhow::Result<crate::BalanceDetails> {
-    let mut interval = crate::INTERVAL;
-    let mut retry_attempt = 0usize;
-
-    loop {
-        if retry_attempt == crate::RETRY_COUNT {
-            anyhow::bail!(
-                "Failed to perform query to RPC after {} attempts. Stop trying.\nAccount {}, block_hash {}",
-                crate::RETRY_COUNT,
-                account_id.to_string(),
-                block_hash.to_string()
-            );
-        }
-        retry_attempt += 1;
-
-        match get_balance(account_id, block_hash, balance_cache, json_rpc_client).await {
-            Ok(res) => return Ok(res),
-            Err(err) => {
-                tracing::error!(
-                    target: crate::LOGGING_PREFIX,
-                    "Failed to request account view details from RPC for account {}, block_hash {}.{}\n Retrying in {} milliseconds...",
-                    account_id.to_string(),
-                    block_hash.to_string(),
-                    err,
-                    interval.as_millis(),
-                );
-                tokio::time::sleep(interval).await;
-                if interval < crate::MAX_DELAY_TIME {
-                    interval *= 2;
-                }
-            }
-        }
-    }
-}
-
 async fn get_balance(
```

Comment:
Previous implementation gave the balance for the end of the block
Your proposal gives the balance for the beginning of the block

Reply:
Discussed offline, this is actually returning the balance for the end of the **previous** block. This name is misleading though so I will rename :)

Reply:
Renamed to `get_balance_before_block`